### PR TITLE
Run exports

### PR DESCRIPTION
**Pull request by wolfv**
this builds on top of the index json PR and adds more complete parsing for `run_exports.json` from packages. 

Also uses that in the channeldata parsing.

### CODE REVIEW

**Comment by wolfv** [comment]
Although I was wondering if we should deserialize to `Vec<String>` or `Vec<MatchSpec>` here and in `index.json`?

File: crates/rattler_conda_types/src/run_export.rs [hunk 1]
```diff
@@ -5,14 +5,7 @@ use serde::{Deserialize, Serialize};
 pub enum RunExportKind {
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/run_export.rs:5]: Could you also add documentation here about what they mean? Or refer to the docs of the other struct?

File: crates/rattler_conda_types/src/run_export.rs [hunk 2]
```diff
@@ -1,18 +1,13 @@
 use serde::{Deserialize, Serialize};
 
 /// Type of runtime export.
+/// See more info in [the conda docs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements)
+/// [`package::run_exports::RunExports`]
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/run_export.rs:5]:
```suggestion
/// [`crate::package::RunExports`]
```

File: crates/rattler_conda_types/src/package/run_exports.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, skip_serializing_none};
+
+/// A representation of the `run_exports.json` file found in package archives.
+///
+/// The `run_exports.json` file contains information about the run exports of a package
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Hash)]
+pub struct RunExports {
```
Comment (baszalmstra) [inline crates/rattler_conda_types/src/package/run_exports.rs:12]: Oh I forgot! Should we call this `RunExportsJson`?

Comment (wolfv) [reply crates/rattler_conda_types/src/package/run_exports.rs:12]: Let's do that later for all types? So far this was just moved.

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**